x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexCompatibility.java

```diff
@@ -35,12 +35,12 @@ public static Map<String, EsField> compatible(Map<String, EsField> mapping, Vers
         return mapping;
     }
 
-    public static EsIndex compatible(EsIndex esIndex, Version version) {
+    public static EsIndex compatible(EsIndex esIndex, TransportVersion version) {
```

Comment:
Either version that's been tracking Version should do (i.e. we have the mapping of the released `Version`s).
`TransportVersion` has been [used previously](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/VersionCompatibilityChecks.java#L24), and it made sense to me, as Version and SqlVersion have been gating features (like date nanos and unsigned long support) and can be seed closer to client serialisation concerns (so "transport"), rather than index concerns.
But ultimately, I think either would work.